H/HDF5/build_tarballs.jl

```diff
@@ -286,12 +298,13 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     # To ensure that the correct version of libgfortran is found at runtime
-    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
-    Dependency("LibCURL_jll"; compat="7.73,8"),
-    Dependency("OpenSSL_jll"; compat="3.0.8"),
-    Dependency("Zlib_jll"),
-    # Dependency("dlfcn_win32_jll"; platforms=filter(Sys.iswindows, platforms)),
-    Dependency("libaec_jll"),   # This is the successor of szlib
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"), v"1.1.1";
+               compat="1.1"),
+    Dependency("LibCURL_jll", v"8.7.1"; compat="7, 8"),
+    Dependency("OpenSSL_jll", v"3.0.13"; compat="3"),
```

Comment:
Again, don't use build version and compat bound, the build version argument is a positional one while the compat one is a keyword for historical reasons, I wish it was the opposite, I really see very little uses for an explicit build version different from the minimum compat bound (which is the default when only the compat is specified)
```suggestion
    Dependency("OpenSSL_jll""; compat="3.0.8"),
```

Reply:
It seems there is a "magic list" that specifies which compat bounds to use for what libraries...